sdk/core/Azure.Core/api/Azure.Core.net461.cs

```diff
@@ -678,6 +678,14 @@ public abstract partial class TokenCredential
         public string[] Scopes { get { throw null; } }
         public string? TenantId { get { throw null; } }
     }
+    public partial class UserAgentValue
+    {
+        public UserAgentValue(System.Type packageType, string? applicationId = null) { }
+        public string? ApplicationId { get { throw null; } }
+        public System.Type PackageType { get { throw null; } }
```

Comment:
This property and naming is a bit confusing - is it meant to be the type that the user used to create the UserAgentValue?

Reply:
>  is it meant to be the type that the user used to create the UserAgentValue?

Yes.

Reply:
I wonder if it would be better to just take an Assembly in the constructor and have an Assembly property. The type used to create the UserAgentValue is somewhat arbitrary as we only use the containing assembly for the string.

Reply:
I reluctantly agree 😄 - I just think the usage of the API where callers need to get the assembly from the type themselves is not great. 

@tg-msft @KrzysztofCwalina  - thoughts?

Reply:
I like `Assembly` too.  `typeof(Foo).Assembly` isn't an undo burden for library authors.